apps/website/src/components/content/YouTubeLightbox.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useEffect, useId, useMemo } from "react";
+import PhotoSwipeLightbox from "photoswipe/lightbox";
+
+import { getDefaultPhotoswipeLightboxOptions } from "@/utils/photoswipe";
+import { camelToKebab } from "@/utils/string-case";
+import { type HTMLAttributes, htmlToReact } from "@/utils/attrs";
+
+const iframeSrc = (id: string) =>
+  `https://www.youtube-nocookie.com/embed/${encodeURIComponent(
+    id
+  )}?modestbranding=1`;
+
+const iframeAttrs: HTMLAttributes = {
+  referrerpolicy: "no-referrer",
+  allow: "encrypted-media",
+  sandbox: "allow-same-origin allow-scripts",
+  loading: "lazy",
+  class: "aspect-video w-full rounded-2xl shadow-xl bg-alveus-green-800",
+};
+
+const safeJsonParse = (str: string) => {
+  try {
+    return JSON.parse(str);
+  } catch {
+    return null;
+  }
+};
+
+const parseUrl = (url: string) => {
+  const match = url.match(
+    /^https:\/\/www\.youtube\.com\/watch\?v=([a-zA-Z0-9_-]+)$/
+  );
+  if (!match?.[1]) throw new Error(`Invalid YouTube URL: ${url}`);
+  return match[1];
+};
+
+type TriggerProps = {
+  videoId: string;
+  caption?: string;
+  className?: string;
+  children: React.ReactNode;
+};
+
+const getTrigger = (id: string) => {
```

Comment:
nitpick but i would suggest naming this `createTrigger` to make clear its a factory